### DISCUSSION TITLE
Selection of multiple items

### DISCUSSION THREAD

**Discussion by Bumblebee001** [GrandOrgue/GoOdf#44] [Ideas]
It would help enormously if it were possible to select several items (eg using Ctrl or Shift keys) to drag, add or remove items in bulk rather than one at a time. Currently some work is very laborious when multiple elemements need to be treated the same way.

**Comment by larspalo** [comment]
I've already begun adding such possibilities with 8ef47b03a3e44d6390b19102d62a63f9a1916dab that takes care of the Switch assignments for most elements - that commit doesn't include that option for the divisionals and generals though, I might add that later.

**Reply by Bumblebee001** [reply]
another way perhaps is to right click on a window and select copy (ctrl+c) and paste (ctrl+v) eg for the selection of elements for addition general pistons....

**Reply by larspalo** [reply]
How do you mean? Copy a selected element from the tree to a new element starting out with identical properties? Copy referenced switches from one selected element to other ones that exists already?

My first intent is just to allow free, multiple, selections for more elements like on the adding of organ elements as GUI Elements on a panel - starting out with Switches as they are quite frequently needed with modern sample sets.

The usage of the old style, odf type, generals (and divisionals) is rather to be reserved for very special cases - just like in the sample set you're working on right now. Any "general" use case for divisionals and generals should be done by setter elements like I describe in https://www.youtube.com/watch?v=nitHpBEDsFI&list=PLNz9AJhjubTdw0pV2nZC_XfphJhJV5ly3&index=6 instead as they are much more flexible and easy to use.

**Reply by Bumblebee001** [reply]
eg to create several couplers with the same referenced stops and couplers

correction: I meant General pistons

**Reply by larspalo** [reply]
> several couplers with the same referenced stops and couplers

Ouch! I don't understand it like that... Couplers cant really reference stops and other couplers. They do behave like any other drawstop type though, and can reference Switches.

Or do you mean "create Generals with the same referenced elements"? In that case see my comment above and watch the screencast.

**Reply by Bumblebee001** [reply]
Please remember..... in your video you have just a few stops and you had to identify them individually as you are referencing switches not actual stops...... Now I have around 35 (H) +35 (I) + 35 (II) + 9 (P) switches that makes up to 114 items to select for each general piston.

**Reply by Bumblebee001** [reply]
My organ is turning out to be a lot more complicated with selectors etc which are not discussed in the video.

**Reply by larspalo** [reply]
> Now I have around 35 (H) +35 (I) + 35 (II) + 9 (P) switches that makes up to 114 items to select for each general piston.

Then you're doing things wrong. The general(s) should _only_ have the H, I, II and Tutti Switches - in the future perhaps a Cresc. So there shouldn't be more than four of the generals at the moment with only four referenced switches each.
